enhancements/network/ovs-hardware-offload.md

```diff
@@ -0,0 +1,225 @@
+configuration, this probably requires a node rebooting to guarantee the order
+if linux bond configuration is applied upfront.
+
+- When enabling OVS `hw-offload` option for OpenvSwitch daemon, it is required
+that ovsdb is created first.
+
+### Risks and Mitigations
+
+RHEL-8.3 contains kernel and driver changes to fully support OVS Hardware
+Offload with connection track feature. If RHEL-8.3 is not used for RHCOS in
+4.6, OVS Hardware Offload feature will not be fully available.
+
+### Test Plan
+
+- The changes in each relevant component must pass full e2e suite.
+- Origin e2e tests mast pass on bare metal with OVS Hardware Offload enabled
+in OpenvSwitch daemon.
+- OVS hardware offload function must work on supported NICs.
+
+### Graduation Criteria
+
+To Tech Preview:
+- Bare metal cluster can be brought up when hardware offload is enabled.
+- A mixed of pods using offload and not using offload can co-exist on
+different nodes and communicate with each other.
+- Network policies can be applied to block pod-to-pod, service-to-service
+and pod-to-service traffic.
+
+### Upgrade / Downgrade Strategy
+
+- Upgrading an existing cluster to a version that supports OVS Hardware Offload
+doesn't require any change in configurations, API use or invocations.
+
+- All the OVS Hardware Offload configuration will need to be removed before
+downgrading to a version that doesn't support OVS Hardware Offload.
+
+### Version Skew Strategy
+
+- OVS hardware offload is an addon feature to existing components that involved.
+The upgrade tests will be conducted in each individual component.
+
+- In a rolling update, some nodes will have the current version and some will
+have the new version. Since all nodes interact with the master and OVS hardware
+offload is a node specific configuration, version skew can occur between nodes.
+
+- The update of Multus CNI config file doesn't affect the use of customized
+net-attach-def for default cluster network.
+
+## Implementation History
+
+- OpenvSwitch package [version 2.13](https://github.com/openshift/ovn-kubernetes/pull/122) is currently used in OpenShift
+- Relevant patches in `ovn-k8s-cni-overlay` CNI plugin are available in OpenShift OVN-Kubernetes
+- Multus is in GA support since 4.1 release and contains relevant patches for specifying customized default net-attach-def
+- [SR-IOV Operator](https://docs.openshift.com/container-platform/4.3/networking/hardware_networks/about-sriov.html) is in GA support since 4.3 release
+
+
+## Drawbacks
+
+OVS Hardware Offload leverages TC flower and actions that are available in the
+linux kernel for traffic classification. When OVS Hardware Offload is enabled,
+any future component that needs apply TC rules directly on primary NIC that
+used by OVS for hardware offloading may cause potential issue by setting
+conflicted rules.
+
+## Alternatives
+
+Enable OVS-DPDK and use the DPDK generic flow interface (rte_flow) for OVS
```

Comment:
is OVS-DPDK hardware offload mature enough for this to be an alternative ?

Reply:
not yet afaik. it is listed as alternative meaning we're not proposing it :-)

Reply:
I understand, so if its not really a valid alternative perhaps it should not be mentioned.
but anyway, it's just nuances...